Content/WorldGeneration/ChestLootSystem.cs

```diff
@@ -0,0 +1,246 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Terraria;
+using System.Text;
+using System.Threading.Tasks;
+using Terraria.ID;
+using Terraria.ModLoader;
+using StarlightRiver.Content.Items.Vitric;
+using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Content.Items.Misc;
+using StarlightRiver.Helpers;
+
+namespace StarlightRiver.Content.WorldGeneration
+{
+    public static class ChestLootSystem
+    {
+        private static Dictionary<ChestRegionFlags, List<ChestLootInfo>> RegionLootInfo;
+        private static Dictionary<ChestRegionFlags, List<ChestLootInfo>> RegionExclusiveLootInfo;
+        private static Dictionary<int, ChestRegionFlags> FramingToRegion;
+        //private const float displayCaseChance = 0.125f;
+
+        public static void Initialize()
+        {
+            FramingToRegion = new Dictionary<int, ChestRegionFlags>
+            {
+                [0] = ChestRegionFlags.Surface,
+                [36] = ChestRegionFlags.Underground,
+                [396] = ChestRegionFlags.Ice,
+                [288] = ChestRegionFlags.Jungle,
+                [360] = ChestRegionFlags.JungleShrine,
+                [576] = ChestRegionFlags.Temple,
+                [468] = ChestRegionFlags.Sky,
+                [612] = ChestRegionFlags.Underwater,
+                [540] = ChestRegionFlags.Spider,
+                [1800] = ChestRegionFlags.Granite,
+                [1836] = ChestRegionFlags.Marble,
+                [144] = ChestRegionFlags.Underworld,
+                [72] = ChestRegionFlags.Dungeon,
+                [432] = ChestRegionFlags.Livingwood,
+                [1152] = ChestRegionFlags.Mushroom,
+                [180] = ChestRegionFlags.Barrel,
+                [216] = ChestRegionFlags.Trashcan,
+                [648] = ChestRegionFlags.Biome,// Jungle
+                [684] = ChestRegionFlags.Biome,// Corruption
+                [720] = ChestRegionFlags.Biome,// Crimson
+                [756] = ChestRegionFlags.Biome,// Hallowed
+                [792] = ChestRegionFlags.Biome,// Ice
+                [2432] = ChestRegionFlags.Biome,// Desert
+                [2144] = ChestRegionFlags.TrappedUnderground,
+                [2360] = ChestRegionFlags.Desert,
+
+            };
+
+            RegionLootInfo = new Dictionary<ChestRegionFlags, List<ChestLootInfo>>();
+
+            foreach (ChestRegionFlags val in Enum.GetValues<ChestRegionFlags>())
+                RegionLootInfo.Add(val, new List<ChestLootInfo>());
+
+            RegionExclusiveLootInfo = new Dictionary<ChestRegionFlags, List<ChestLootInfo>>();
+
+            foreach (ChestRegionFlags val in Enum.GetValues<ChestRegionFlags>())
+                RegionExclusiveLootInfo.Add(val, new List<ChestLootInfo>());
+        }
+
+        public static void Load()
+        {
+            Initialize();
+
+            AddLoot(ModContent.ItemType<BarbedKnife>(), ChestRegionFlags.Surface | ChestRegionFlags.Barrel);
```

Comment:
Having this all be in a single method in a single file really isnt great, it would be nice to have a way to load these across files. One per chest region would likely be a sane design pattern.

Reply:
I decided on there being one file per folder instead of one per chest region (as many items can often be in multiple regions)